ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,273 @@
+        
+        nongray_data(nongray_data ~= 0) =1;
+
+        NonGrey=nongray_data;
+        NonGreyPlusEdge=NonGrey+edge_data;
+        Grey=(NonGreyPlusEdge-1)*-1;
+        VolSmoothROIs_data=wmparc_data.*0;
+        VolSmoothROIs_data(NonGrey==1)=1;
+        VolSmoothROIs_data(Grey==1)=2;
+        VolSmoothROIs_data(edge_data==1)=3;
+        VOLSMOOTHROIS=WMPARC.*0;
+        VOLSMOOTHROIS(std(VOLUME,[],2)>0)=VolSmoothROIs_data;
+        VOLSMOOTHROIS=reshape(VOLSMOOTHROIS,size(SBREFOrig,1),size(SBREFOrig,2),size(SBREFOrig,3),size(SBREFOrig,4));
+        save_avw(VOLSMOOTHROIS,[SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz'],'f',[1.6 1.6 1.6 1]);
+        system(['fslcpgeom ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_SBRef.nii.gz ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz -d']);
+        
+        SmoothingSigma=FinalSpatialSmoothingFWHM/(2*sqrt(2*log(2)));
+        system(['wb_command -volume-label-import ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz "" ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz']);
+        system(['wb_command -volume-parcel-smoothing ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC.nii.gz ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz ' num2str(SmoothingSigma) ' ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_FWHM' num2str(FinalSpatialSmoothingFWHM) '.nii.gz']);
+
+        VOLUMESMOOTH=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_FWHM' num2str(FinalSpatialSmoothingFWHM) '.nii.gz']);
+        VOLUMESMOOTH=reshape(VOLUMESMOOTH,size(VOLUMESMOOTH,1)*size(VOLUMESMOOTH,2)*size(VOLUMESMOOTH,3),size(VOLUMESMOOTH,4));
+        VolumeSmooth_data=VOLUMESMOOTH(std(VOLUMESMOOTH,[],2)>0,:);
+        VolumeSmooth_data=VolumeSmooth_data./std(Volume_data(:));
+        CIFTIMSMAllSmooth.cdata=CIFTIMSMAllSmooth.cdata./std(CIFTIMSMAll.cdata(:));
+        
+        % structure idx
+        CIFTI_LENGTH=length(CIFTI.cdata);
+        CORTEX_start_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_LEFT_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_RIGHT_end_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
+
+        CORTEX_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
+        CEREBELLUM_start_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_LEFT').ciftilist);
+        CEREBELLUM_end_idx=max(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_RIGHT').ciftilist);
```

Comment:
Just take the whole index list for these structures, it is fewer and less complicated lines of code:
```suggestion
        CEREBELLUM_idx_list = cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_LEFT').ciftilist;
```
...
```
NewFeatures(k, 10) = sum(abs(CIFTI.cdata(CEREBELLUM_idx_list, k))) ./ sum(abs(Volume_data(:, k))); %Cerebellum/Volume
```

Reply:
Isn't this only the list for CEREBELLUM_LEFT?

Reply:
Yes, I saw that eventually and fixed it.